client/src/app/journey/journey.service.ts

```diff
@@ -47,16 +45,15 @@ export class JourneyService {
     });
   }
 
-  getVersionsByRole(roleId) {
+  getAllJourneyVersions() {
     this.headers.set('X-TOKEN', this.loginService.getSessionId());
-    let url = this.config.Server + 'nsdc/v1.0/journeys/versions/roles/' + roleId;
-    return this.http.get(url, { headers: this.headers })
-    .map(response => response.json());
+    return this.http.get(this.config.Server + 'nsdc/v1.0/journeys/versions', { headers: this.headers })
+    .map(res => res.json() as JourneyVersion[]);
   }
 
-  getVersionById(id) {
+  getVersionsByRole(roleId) {
     this.headers.set('X-TOKEN', this.loginService.getSessionId());
-    let url = this.config.Server + 'nsdc/v1.0/journeys/versions/' + id;
+    let url = this.config.Server + 'nsdc/v1.0/journeys/versions/roles/' + roleId;
```

Comment:
This always bugged, me, why are we creating a variable here, let along a let ... We can look at this when we refactor the services.